hack/debug-network.sh

```diff
@@ -0,0 +1,729 @@
+        die "Could not find node name in ${config_file}"
+    fi
+
+    local lognode="${NODE_DIR}/${node}"
+    mkdir -p "${lognode}"
+
+    # Grab node service logs and config file
+    log_service "${lognode}" "${AOS_NODE_SERVICE}" "node"
+
+    # Log the generic system stuff
+    log_system "${lognode}"
+
+    # Log some node-only information
+    echo_and_eval  brctl show                              &> "${lognode}/bridges"
+    echo_and_eval  docker ps -a                            &> "${lognode}/docker-ps"
+    echo_and_eval  ovs-ofctl -O OpenFlow13 dump-flows br0  &> "${lognode}/flows"
+    echo_and_eval  ovs-ofctl -O OpenFlow13 show br0        &> "${lognode}/ovs-show"
+    echo_and_eval  tc qdisc show                           &> "${lognode}/tc-qdisc"
+    echo_and_eval  tc class show                           &> "${lognode}/tc-class"
+    echo_and_eval  tc filter show                          &> "${lognode}/tc-filter"
+    echo_and_eval  systemctl cat docker.service            &> "${lognode}/docker-unit-file"
+    echo_and_eval  cat "$(systemctl cat docker.service | grep EnvironmentFile.\*openshift-sdn | awk -F=- '{print $2}')" \
+                                                           &> "${lognode}/docker-network-file"
+
+    # Iterate over all pods on this node, and log some data about them.
+    # Remember the name, address, namespace, and pid of the first pod we find on
+    # this node which is not in the default namespace
+    local base_pod_addr
+    while read pod_node pod_nodeaddr pod_name pod_ns pod_addr pod_id; do
+        if [[ "${pod_node}" != ${node} ]]; then
+            continue
+        fi
+
+        local logpod="${lognode}/pods/${pod_name}"
+        mkdir -p "${logpod}"
+
+        local pid
+        pid="$(docker inspect -f '{{.State.Pid}}' "${pod_id}")"
+        if [[ -z "${pid}" ]]; then
+            echo "${node}:${pod_name}: could not find pid of ${pod}"
+            continue
+        fi
+
+        echo_and_eval nsenter -n -t "${pid}" -- ip addr  show  &> "${logpod}/addresses"
+        echo_and_eval nsenter -n -t "${pid}" -- ip route show  &> "${logpod}/routes"
+
+        # If we haven't found a local pod yet, or if we have, but it's
+        # in the default namespace, then make this the new base pod.
+        if [[ -z "${base_pod_addr}" || "${base_pod_ns}" = "default" ]]; then
+            base_pod_addr="${pod_addr}"
+            base_pod_ns="${pod_ns}"
+            base_pod_name="${pod_name}"
+            base_pod_pid="${pid}"
+        fi
+    done < "${META_DIR}/podinfo"
+
+    if [[ "${SKIP_TESTS}" = "true" ]]; then
+        echo "Skipping all pod, service and external connectivity tests as indicated"
+        return
+    fi
+
+    if [[ -z "${base_pod_addr}" ]]; then
+        echo "No pods on ${node}, so no connectivity tests"
+        return
+    fi
+
+    local base_pod_port
+    base_pod_port="$(get_port_for_addr "${base_pod_addr}" "${lognode}")"
+    if [[ -z "${base_pod_port}" ]]; then
```

Comment:
seems like if you're going to do this check on the output of `get_port_for_addr` a lot, you should internalize it to that function and have it return non-zero, so callers just have to check the return code of `get_port_for_addr`